feat(elasticsearch) add elasticsearch to dependency train

Emalton opened the pull request:
**Description**
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
Elasticsearch doesn't need anything besides a port which is similar to memcached.

**⚙️ Type of change**

- [X] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->
Untested, made this based off of memcached configs. 

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [X] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [X] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [X] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [X] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

Emalton committed:
Add elasticsearch to dependency train

Untested, made this based off of memcached configs. Elasticsearch doesn't need anything besides a port which is similar to memcached.

Emalton committed:
Update values.yaml

Add `readOnlyRootFilesystem: false` per recommendation from Steven.

Signed-off-by: John P <emaltonator@gmail.com>

Emalton committed:
Fix version

Fix versions

Emalton committed:
Fix version

Update values.yaml to newer version

Emalton committed:
rookie mistake

redis -> elasticsearch

StevenMcElligott committed:
fix ROFS

StevenMcElligott committed:
run as root

StevenMcElligott committed:
add persistence

StevenMcElligott committed:
mount whole directory

StevenMcElligott committed:
update tag

StevenMcElligott committed:
persist everything and see

StevenMcElligott committed:
another mountpath and remove tcp

StevenMcElligott committed:
test w/o persistance and additional capabilities

StevenMcElligott committed:
needs this for root, but I don't think it's necessary

StevenMcElligott committed:
quick test to see if I can drop down privs

StevenMcElligott reviewed (approved):
I helped with the end of this PR so I'm happy, just need the injector parts

Emalton commented:
Injector portion should be all set on this PR:

truecharts/library-charts#524

stavros-k commented on charts/dependency/elasticsearch/values.yaml:
```diff
@@ -0,0 +1,28 @@
+persistence:
+  data:
+    enabled: true
+    mountPath: "/bitnami/mongodb"
```
I'd suggest going over your PR!
This looks like its not tested at all.

Also, I don't see any inputs for password. (Yes ES has auth)

Emalton replied:
Fixed that, also removed that persistence there too. Not sure how I missed that lol. Thanks for your help with this.

stavros-k replied:
??? So without persistent data, how does that work?

StevenMcElligott replied:
> Fixed that, also removed that persistence there too. Not sure how I missed that lol. Thanks for your help with this.

The chart needs persistence, just copied the wrong one from another bitnami chart, look into it otherwise you lose everything in the container upon restarts

Emalton replied:
Just fixed that, lol I have no idea why I removed that. The path from the docs says `/usr/share/elasticsearch/data` named `data` so I tossed that in there! What an epic fail :)

Emalton committed:
Add Elasticsearch Password

Attempt at adding the elasticsearch password, based on information from the injector (specifically the template file for the library-charts repo.)

Emalton committed:
Re-add persistence

No idea why I removed it but here it is again!

Emalton commented:
Thinking I have everything sorted out here, so going to take a look what needs to be done [here](https://github.com/truecharts/library-charts/pull/524). Appreciate the guidance/help from everyone!

Crow-Control commented:
we're going to hold off on adding an elastic search dependency for now.
